File: MPP/Laborator1/src/test/java/repository/ParticipantRepositoryTest.java
package repository;

import model.Participant;
import org.junit.Assert;
import org.junit.Test;
import org.junit.internal.runners.statements.Fail;

import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;
import java.util.stream.Collectors;
import java.util.stream.StreamSupport;

import static org.junit.Assert.*;


public class ParticipantRepositoryTest {
    IRepositoryParticipant repo;
    Participant participant=new Participant("Teodora",11);
    Participant lastParticipant;

    public ParticipantRepositoryTest() {
        Properties prop=new Properties();

        try {
            prop.load(new FileReader("D:\\University-Courses\\MPP\\Laborator1\\src\\test\\resources\\bd.properties"));
        } catch (IOException e) {
            e.printStackTrace();
        }
       repo=new ParticipantRepository(prop);
    }

    @Test
    public void test() {
        //size
        assertEquals(5,repo.size());

        //save
        repo.save(participant);
        assertEquals(6,repo.size());
        List<Participant> lista=
                StreamSupport.stream(repo.findAll().spliterator(),false )
                        .collect(Collectors.toList());
        lastParticipant=lista.get(lista.size()-1);
        participant.setID(lastParticipant.getID());

        //findone
        assertEquals(participant,repo.findOne(lastParticipant.getID()));

        //update
        repo.update(lastParticipant.getID(),new Participant("Teodora",6));

        lista=
                StreamSupport.stream(repo.findAll().spliterator(),false )
                        .collect(Collectors.toList());
        lastParticipant=lista.get(lista.size()-1);

        participant.setVarsta(6);
        assertEquals(participant,lastParticipant);

        try{
            repo.update(-1,participant);
            fail();
        }
        catch (RepositoryException e){
            assertEquals("Error: Nu s-a putut actualiza participantul!",e.getMessage());
        }
        //delete
        repo.delete(lastParticipant.getID());
        assertEquals(5,repo.size());
        try{
            repo.delete(-1);
            fail();
        }
        catch (RepositoryException e){
            assertEquals("Error: Nu s-a putut sterge participantul!",e.getMessage());
        }
    }

}

File: MPP/LabC#/UnitTest/ParticipantRepositoryTest.cs
using System;
using System.Collections.Generic;
using Concurs.model;
using Concurs.repository.utils;
using log4net.Config;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace UnitTest
{
    [TestClass]
    public class ParticipantRepositoryTest
    {
        [TestMethod]
        public void TestMethod()
        {
            XmlConfigurator.Configure(new System.IO.FileInfo("App.config"));

            IDictionary<String, string> props = new SortedList<String, String>();

            props.Add("ConnectionString", Concurs.repository.DBUtils.GetConnectionStringByName("catalogTest"));

            ParticipantRepository repo = new ParticipantRepository(props);

            Participant participant = new Participant("Teodora", 11);
            Participant lastParticipant;

            //size

            Assert.AreEqual(repo.Size(),5);

            //save
            repo.Save(participant);
            Assert.AreEqual(6, repo.Size());
            List<Participant> lista =(List<Participant>) repo.FindAll();
            lastParticipant = lista[lista.Count - 1];
            participant.Id=lastParticipant.Id;

            //findone
            Assert.AreEqual(participant, repo.FindOne(lastParticipant.Id));
            Assert.IsNull(repo.FindOne(-1));

            //update
            repo.Update(lastParticipant.Id, new Participant("Teodora", 6));

            lista = (List<Participant>)repo.FindAll();
            lastParticipant = lista[lista.Count - 1];

            participant.Varsta=6;
            Assert.AreEqual(participant, lastParticipant);

            try
            {
                repo.Update(-1, participant);
                Assert.Fail();
            }
            catch (RepositoryException e)
            {
                Assert.AreEqual(e.Message, "Error: Nu s-a putut actualiza participantul!");
            }

            //delete
            repo.Delete(lastParticipant.Id);
            Assert.AreEqual(repo.Size(), 5);

            try
            {
                repo.Delete(lastParticipant.Id);
                Assert.Fail();
            }
            catch (RepositoryException e)
            {
                Assert.AreEqual(e.Message, "Error: Nu s-a putut sterge participantul!");
            }
        }
    }
}
